[MIST-1068] Move queryStarter and queryRemover from Group to ApplicationInfo

kyungissac opened the pull request:
The PR addresses #1068.

Closes #1068.

taegeonum commented on mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAwareQueryManagerImpl.java:
```diff
@@ -208,7 +208,7 @@ public Query createAndStartQuery(final String queryId,
     // Start the submitted dag
     // TODO : [MIST-1016] get the appropriate Group for this applicationInfo.
     final Group group = applicationInfo.getRandomGroup();
```
please remove this line

kyungissac commented:
@taegeonum I've addressed your comment. Thanks!

taegeonum commented on mist-core/src/main/java/edu/snu/mist/core/task/checkpointing/DefaultCheckpointManagerImpl.java:
```diff
@@ -178,7 +178,7 @@ public void deleteGroup(final String groupId) {
       return;
     }
-    group.getQueryRemover().deleteAllQueries();
+    group.getApplicationInfo().getQueryRemover().deleteAllQueries();
```
This will delete all queries in the application.

kyungissac commented:
@taegeonum I've addressed your comment. Thanks!

Due to this change, the method `deleteAllQueries()` in the `QueryRemover` interface is currently not used anywhere. Do you think we should remove it, or should we keep it in case we delete an app?

taegeonum commented:
@kyungissac I think it would be good to keep the interface now.  I will merge this PR. Thanks!
